benchmarks/c/miscellaneous/pthread.c

```diff
@@ -0,0 +1,413 @@
+        cond_timedwait(&cond, &cond_mutex, 10L);
+        idle = phase < 2;
+        mutex_unlock(&cond_mutex);
+    }
+    return message;
+}
+
+void cond_test()
+{
+    void* message = (void*) 42;
+    mutex_init(&cond_mutex, PTHREAD_MUTEX_NORMAL, PTHREAD_PRIO_NONE, PTHREAD_MUTEX_POLICY_FIRSTFIT_NP, 0);
+    cond_init(&cond);
+
+    pthread_t worker = thread_create(cond_worker, message);
+
+    {
+        mutex_lock(&cond_mutex);
+        ++phase;
+        cond_signal(&cond);
+        mutex_unlock(&cond_mutex);
+    }
+
+    {
+        mutex_lock(&cond_mutex);
+        ++phase;
+        cond_broadcast(&cond);
+        mutex_unlock(&cond_mutex);
+    }
+
+    void* result = thread_join(worker);
+    assert(result == message);
+
+    cond_destroy(&cond);
+    mutex_destroy(&cond_mutex);
+}
+
+// -------- reader/writer locks
+
+//from pthread.h for darwin:
+//define PTHREAD_PROCESS_SHARED 0
+//define PTHREAD_PROCESS_PRIVATE 1
+void rwlock_init(pthread_rwlock_t* lock, int shared)
+{
+    int status;
+    int value;
+    pthread_rwlockattr_t attributes;
+    status = pthread_rwlockattr_init(&attributes);
+    assert(status == 0);
+
+    status = pthread_rwlockattr_setpshared(&attributes, shared);
+    assert(status == 0);
+    status = pthread_rwlockattr_getpshared(&attributes, &value);
+    assert(status == 0);// && value == shared); //TODO Add storage of attribute entries.
+
+    status = pthread_rwlock_init(lock, &attributes);
+    assert(status == 0);
+    status = pthread_rwlockattr_destroy(&attributes);
+    assert(status == 0);
+}
+
+void rwlock_destroy(pthread_rwlock_t* lock)
+{
+    int status = pthread_rwlock_destroy(lock);
+    assert(status == 0);
+}
+
+void rwlock_wrlock(pthread_rwlock_t* lock)
+{
+    int status = pthread_rwlock_wrlock(lock);
+    assert(status == 0);
+}
+
+bool rwlock_trywrlock(pthread_rwlock_t* lock)
+{
+    int status = pthread_rwlock_trywrlock(lock);
+    //assert(status == 0 || status == EBUSY); //TODO Add support for platform-dependent error codes.
+    return status == 0;
+}
+
+void rwlock_rdlock(pthread_rwlock_t* lock)
+{
+    int status = pthread_rwlock_rdlock(lock);
+    assert(status == 0);
+}
+
+bool rwlock_tryrdlock(pthread_rwlock_t* lock)
+{
+    int status = pthread_rwlock_tryrdlock(lock);
+    //assert(status == 0 || status == EBUSY); //TODO Add support for platform-dependent error codes.
+    return status == 0;
+}
+
+void rwlock_unlock(pthread_rwlock_t* lock)
+{
+    int status = pthread_rwlock_unlock(lock);
+    assert(status == 0);
+}
+
+void rwlock_test()
+{
+    pthread_rwlock_t lock;
+    rwlock_init(&lock, PTHREAD_PROCESS_PRIVATE);
+    int const test_depth = 4;
+
+    {
+        rwlock_wrlock(&lock);
+        bool success = rwlock_trywrlock(&lock);
+        assert(!success);
+        success = rwlock_tryrdlock(&lock);
+        assert(!success);
+        rwlock_unlock(&lock);
+    }
+
+    {
+        __VERIFIER_loop_bound(test_depth);
+        for (int i = 0; i < test_depth; i++)
```

Comment:
You need `test_depth + 1` in the bound annotation (same for the other loop below).